webapp/src/components/systemConsole/questions/questions.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import './style.scss';
+import {useDebouncedCallback} from 'use-debounce';
+import utils from 'utils/utils';
+
+import type {CustomSettingChildComponentProp} from 'components/systemConsole/index';
+
+import type {SurveyQuestionsConfig} from 'types/mattermost-webapp';
+
+export type QuestionType = 'linear_scale' | 'text';
+
+const questionTypeDisplayName = new Map<QuestionType, string>([
+    ['linear_scale', 'Linear scale question (1 to 10)'],
+    ['text', 'Textual question'],
+]);
+
+export type Question = {
+    id: string;
+    title?: string;
+    text?: string;
+    type: QuestionType;
+    system: boolean;
+    mandatory: boolean;
+    helpText?: string;
+};
+
+function SurveyQuestions({id, setSaveNeeded, onChange, config, setInitialSetting}: CustomSettingChildComponentProp) {
+    const [questions, setQuestions] = useState<Question[]>([]);
+
+    const generateDefaultQuestions = (): Question[] => {
+        return [
+            {
+                id: utils.uuid(),
```

Comment:
Why `uuid` out of curiosity? Do we just want a unique id here for these? If so, could we just hardcode them as default somehow? Or is there some validation where we want these to be UUIDs?

Reply:
Just need something nice and random. This will also be needed when we add the functionality to add custom questions.